QubitServer/src/main/java/org/labrad/qubits/channeldata/AnalogDataTime.scala

```diff
@@ -0,0 +1,68 @@
+package org.labrad.qubits.channeldata
+
+import java.util.concurrent.Future
+
+import org.labrad.qubits.channels.AnalogChannel
+import org.labrad.qubits.proxies.DeconvolutionProxy
+import org.labrad.qubits.util.Futures
+
+import com.google.common.base.Function
+
+class AnalogDataTime(data: Array[Double], isDeconvolved: Boolean, averageEnds: Boolean, dither: Boolean) extends AnalogDataBase {
+
+  private var deconvolvedData: Array[Int] = null
+
+  if (isDeconvolved) {
+    deconvolvedData = data.map { x =>
+      (x * 0x1fff).toInt & 0x3fff
+    }
+  }
+  setDeconvolved(isDeconvolved)
+
+  def checkLength(expected: Int): Unit = {
+    LengthChecker.checkLengths(data.length, expected)
+  }
+
+  override def deconvolve(deconvolver: DeconvolutionProxy): Future[Void] = {
+    val ch = getChannel()
+    val req = deconvolver.deconvolveAnalog(
+        ch.getDacBoard(),
+        ch.getDacId(),
+        data,
+        ch.getSettlingRates(),
+        ch.getSettlingTimes(),
+        ch.getReflectionRates(),
+        ch.getReflectionAmplitudes(),
+        averageEnds,
+        dither)
+    Futures.chain(req, new Function[Array[Int], Void] {
+      override def apply(result: Array[Int]): Void = {
+        deconvolvedData = result
+        setDeconvolved(true)
+        null
+      }
+    })
+  }
+
+  override def getDeconvolved(): Array[Int] = {
+    require(isDeconvolved(), "Data has not yet been deconvolved")
+    deconvolvedData
+  }
+}
+
+class AnalogDataTimeDacified(data: Array[Int]) extends AnalogDataBase {
```

Comment:
yup :)

scala completely decouples the file name from the class names, so you have to be a little bit careful to name things in a sane way, but it's very convenient to collect small related classes into a single file. I find it makes comprehending the larger structure of the code much easier than when every class is in a different file.